test/e2e/specs/editor/various/keep-styles-on-block-transforms.spec.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Keep styles on block transforms', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'Should keep colors during a transform', async ( {
+		page,
+		editor,
+	} ) => {
+		await page.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '## Heading' );
+
+		await editor.openDocumentSettingsSidebar();
+		await page.click( 'role=button[name="Text"i]' );
+
+		await page.click( 'role=button[name="Color: Luminous vivid orange"i]' );
+
+		await page.click( 'role=button[name="Heading"i]' );
+		await page.click( 'role=menuitem[name="Paragraph"i]' );
+
+		expect( await editor.getEditedPostContent() )
```

Comment:
The recommended approach now is to use `editor.getBlocks` and `expect().toMatchObject` as mentioned in the doc. Have you tried those?

Reply:
I tried but didn't work, still will try again and update if it works